### ISSUE TITLE
Test-Connection fails when run as root on MacOS

### ISSUE THREAD

**Issue by tipzie** [PowerShell/PowerShell#13238]
## Steps to reproduce
1. Install Powershell Core from .pkg
2. run 'sudo pwsh' to run powershell as root
3. Try a Test-Connection to any IP address. 

## Expected behavior

```none
Test-Connection attempts to connect to the IP and returns the results of the connection attempt.
```
![image](https://user-images.githubusercontent.com/68653255/88195958-a2bb1700-cc0e-11ea-9ace-57df88e34a73.png)

## Actual behavior
```none
Test-Connection errors instantly with the following error message and nothing is returned. Error is still instant even if a timeout is specified.
```
![image](https://user-images.githubusercontent.com/68653255/88193169-59b59380-cc0b-11ea-8b23-b2b3cc2f02bb.png)

## Environment data

```none
Name                           Value
----                           -----
PSVersion                      7.0.3
PSEdition                      Core
GitCommitId                    7.0.3
OS                             Darwin 18.7.0 Darwin Kernel Version 18.7.0: Tue Aug 20 16:57:14 PDT 2019; root:xnu-4903.271.2~2/RELEASE_X86_64
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by vexx32** [comment]
Can you call `Get-Error` after this and post the result?

**Comment by tipzie** [comment]
Sure thing
<img width="1093" alt="Screen Shot 2020-07-22 at 12 13 06 PM" src="https://user-images.githubusercontent.com/68653255/88200952-d9942b80-cc14-11ea-9cb7-8b9ddfa78d1c.png">

**Comment by daxian-dbw** [comment]
@tipzie It may be already implied, but just so I'm clear, does `Test-Connection` work when you run pwsh as a regular user instead of sudo?

**Comment by tipzie** [comment]
Yup, runs as expected on non-root. That's the output I posted under Expected Behavior.

**Comment by iSazonov** [comment]
@tipzie Please check with latest PowerShell 7.1 Preview.

**Comment by tipzie** [comment]
Works as expected with 7.1 Preview

**Comment by vexx32** [comment]
Sounds like there might have been an issue in .NET core that has since been fixed? 🤔

**Comment by iSazonov** [comment]
It comes from .Net. I do not think it makes sense to ask the fix for .Net 3 because we already moved to .Net 5.0. Also the cmdlet is an experimental and it is not ready for production (there are many other issues in the .Net API).

**Comment by daxian-dbw** [comment]
@tipzie Thanks for checking the preview version!
The root cause looks like this one: https://github.com/dotnet/runtime/issues/34879 and it's fixed in 5.0.
The labels on the PR don't suggest it's going to be back ported to 3.1.
